apps/src/code-studio/initApp/project.js

```diff
@@ -1462,13 +1471,21 @@ var projects = (module.exports = {
       }
 
       if (JSON.stringify(currentSources) === JSON.stringify(newSources)) {
-        hasProjectChanged = false;
+        if (!projectChangedWhileSaveInProgress) {
+          hasProjectChanged = false;
+        }
+        projectSaveInProgress = false;
+        projectChangedWhileSaveInProgress = false;
```

Comment:
(Sorry for commenting after the merge)
Is this section needed? I'm not too familiar with how currentSources is populated - does the above if condition definitively check whether sources changed?

Reply:
do you mean is it needed in general or specifically for Javalab projects? I think it may be more relevant for other labs

Reply:
dug into this a bit. It gets set when we call saveSources (whether or not the sources successfully saved)